modules/mudletdiscord.go

```diff
@@ -0,0 +1,166 @@
+package modules
+
+import (
+	"embed"
+	"strconv"
+
+	"github.com/GoMudEngine/GoMud/internal/configs"
+	"github.com/GoMudEngine/GoMud/internal/events"
+	"github.com/GoMudEngine/GoMud/internal/plugins"
+	"github.com/GoMudEngine/GoMud/internal/users"
+	"github.com/GoMudEngine/GoMud/internal/mudlog"
+)
+
+var (
+	//go:embed mudletdiscord/*
+	mudletdiscord_Files embed.FS
+)
+
+func init() {
+	// Create the module instance first
+	g := MudletDiscordModule{
+		plug: plugins.New(`mudletdiscord`, `1.0`),
+	}
+
+	// Attach the file system to the plugin
+	if err := g.plug.AttachFileSystem(mudletdiscord_Files); err != nil {
+		panic(err)
+	}
+
+	// Set default values directly during initialization
+	g.setDefaultValues()
+
+	// Register the onLoad callback to load config values after plugin system is ready
+	g.plug.Callbacks.SetOnLoad(func() {
+		g.RefreshConfig()
+	})
+
+	// Export a function to register event listeners for any user with GMCP support
+	// The function name remains "RegisterMudletUser" for backward compatibility
+	g.plug.ExportFunction("RegisterMudletUser", g.registerEventListeners)
+}
+
+type MudletDiscordModule struct {
+	plug          *plugins.Plugin
+	inviteUrl     string
+	applicationId string
+	largeImageKey string
+	state         string
+	details       string
+	mudName       string
+	// Track which users already have event listeners registered
+	registeredUsers map[int]bool
+}
+
+// Set hardcoded default values for initialization
+func (g *MudletDiscordModule) setDefaultValues() {
+	g.mudName = "GoMud"
+	g.inviteUrl = "https://discord.gg/FaauSYej3n"
+	g.applicationId = "1234"
+	g.largeImageKey = "server-icon"
+	g.state = "Playing GoMud"
+	g.details = "using GoMud Engine"
```

Comment:
These should all be defined in `/data-overlays/config.yaml`, no need to define it here.

It's read-only, so any changes via `server set` etc. would just copy it to the overrides... until that happens, the data-overlays/config.yaml would be used.